memtable.cc

```diff
@@ -710,7 +714,10 @@ memtable::apply(const mutation& m, db::rp_handle&& h) {
           with_linearized_managed_bytes([&] {
             auto& p = find_or_create_partition(m.decorated_key());
             _stats_collector.update(*m.schema(), m.partition());
-            p.apply(*_schema, m.partition(), *m.schema());
+            mutation_application_stats app_stats;
+            p.apply(*_schema, m.partition(), *m.schema(), app_stats);
+            _table_stats.memtable_row_insertions += app_stats.row_insertions;
+            _table_stats.memtable_row_hits += app_stats.row_hits;
```

Comment:
Not a must, but you could avoid some code duplication if you stored `_table_stats.memtable` as `mutation_application_stats`, and defined operator+= on `mutation_application_stats`. Then you could just do `_table_stats.memtable += app_stats` here and below and don't care what stats are exactly tracked. So slightly more future proof.

Reply:
Good idea - converted those two fields into `mutation_application_structs memtable;` and added `+=` operator. I named it `memtable_app_stats` in order not to clash with other `table_stats` members that have names with `memtable_*` prefix.